docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+        }
+    )
+    df
+
+``kartothek`` allows users to (physically) partition their data by the
+values of table columns such that all the rows with the same value of the column all get
+written to the same partition. To do this, we use the ``partition_on`` keyword argument:
+
+.. ipython:: python
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "partitioned_dataset",
+        df,
+        partition_on = 'E',
+    )
+    dm
+
+
+Of interest here is ``dm.partitions``:
+
+.. ipython:: python
+
+    dm.partitions
+
+
+We can see that partitions have been stored in a way which indicates the
+specific value for the column on which partitioning has been performed.
+
+Partitioning can also be performed on multiple columns; in this case, columns should
+be specified as a list:
+
+.. ipython:: python
+
+    duplicate_df = df.copy()
+    duplicate_df.F = "bar"
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_partitioned_dataset",
+        [df, duplicate_df],
+        partition_on = ['E', 'F'],
+    )
+    dm.partitions
+
+
+Note that, since 2 dataframes have been provided as input to the function, there are
+4 different files created, even though only 2 different combinations of values of E and
+F are found, ``E=test/F=foo`` and ``E=train/F=foo`` .
+(However, these 4 physical partitions can be read as just the 2 logical partitions by
+using the argument ``concat_partitions_on_primary_index=True`` at reading time).
+
+For datasets consisting of multiple tables, partitioning on
+columns only works if the column exists in both tables and is of the same data type.
+
+For example:
+
+.. ipython:: python
+
+    df.dtypes
+    different_df = pd.DataFrame(
+        {
+            "B": pd.to_datetime(["20130102","20190101"]),
+            "L": [1, 4],
+            "Q": [True, False],
+        }
+    )
+    different_df.dtypes
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "multiple_partitioned_tables",
+        [{"data": {"table1": df, "table2": different_df}}],
+        partition_on='B',
+    )
+
+    dm.partitions
+
+
+Because partitions are physical in nature, it is not possible to modify the
+partitioning scheme of an existing dataset via an update, instead, the dataset
+would have to be re-created.
+
+.. note:: Under the hood, partitions are structurally identical to each other and each partition
+    is made up of a collection of files containing the subset of data of each table
+    belonging to that partition.
+
+Secondary Indices
+-----------------
+
+The ability to build and maintain secondary indices are an additional ability
+provided by ``kartothek``. Secondary indices are `similar` to partitions in the
+sense that they allow faster access to subsets of data. The main difference
+between them is that while partitioning actually creates separate partitions based
+on column values, secondary indices are simply python dictionaries mapping column
```

Comment:
Technically speaking the `PartitionIndex` which is created above is also just a dictionary in the end.

Instead of describing the data type you should describe what the index is all about: For every unique value of a given column over the entire dataset, we store all references to the partitions where this value occurs.

What makes the secondary index special opposed to the `PartitionIndex` above is that it allows for N:1 mappings (N unique values in one partition) while the PartitionIndex is a 1:1 relation